src/TRestAxionXrayWindow.cxx

```diff
@@ -0,0 +1,318 @@
+ *                                                                       *
+ * REST is free software: you can redistribute it and/or modify          *
+ * it under the terms of the GNU General Public License as published by  *
+ * the Free Software Foundation, either version 3 of the License, or     *
+ * (at your option) any later version.                                   *
+ *                                                                       *
+ * REST is distributed in the hope that it will be useful,               *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the          *
+ * GNU General Public License for more details.                          *
+ *                                                                       *
+ * You should have a copy of the GNU General Public License along with   *
+ * REST in $REST_PATH/LICENSE.                                           *
+ * If not, see http://www.gnu.org/licenses/.                             *
+ * For the list of contributors see $REST_PATH/CREDITS.                  *
+ *************************************************************************/
+
+//////////////////////////////////////////////////////////////////////////
+/// TRestAxionXrayWindow implements parameters that define the windows
+/// properties, such as material and thickness. This class will load the
+/// transmission data to calculate the transmission for a photon of a
+/// given energy and position. The window might be defined as a uniform
+/// foil or using a particular structure.
+///
+/// For the moment, the window geometry is fixed to be a circular window.
+/// Therefore, the following are the most basic parameters of any window
+/// construction:
+///
+/// * **material**: The material used by the window. The name must match
+/// the name of one of the files found at `data/transmission/` removing
+/// the extension.
+/// * **thickness**: The depth of the window that defines finally the total
+/// x-ray photon absorption.
+/// * **radius**: The radius of the window where photons will be accepted.
+///
+/// On top of that, we might define different types of windows that will
+/// allow us to create different structures, such as a strong back where
+/// photon opacity is only defined at a regular grid or stripped pattern.
+/// We may define the window type using:
+///
+/// - **type**: We may define different options that define the regions
+/// of space where the window material will be effective.
+///   -# **foil**: It defines a homogeneous region of material, all the
+///    region inside the window radius will present the same efficiency.
+///   -# **stripped**: It defines a stripped pattern where the absorption
+///   will be effective, otherwise, if the photon is inside the window
+///   but it doesnt hit the stripped pattern, the transmission will be
+///   equal to 1.
```

Comment:
In reality a real x-ray window is built using a combination of `TRestXrayWindow`. So, when you define the type *grid* or the type *strip* if the pattern is hitted then we apply the transmission curve using the material the pattern is made of. But if you do not hit the pattern, then the photon just goes through. There is vacuum.

If you build the window as a combination of Si grid + Al foil, then you are combining two `TRestXrayWindow` descriptions. The second one will apply the Al transmission everywhere, since it is a foil type (as soon as the particle falls inside the window).

>> Transmission is 1 if it doesn't hit the pattern? That seems weird.

So, yes, if it doesn't hit the pattern the photon just goes through.

>> Hitting the strips should have a very low (but not 0) transmission. 

Right, the window is built with a particular material and thickness.

>> Hitting the non strip parts should use the window thickness, so transmission somewhere between 0 and 1.

As I tried to explain this is the combination of 2 `TRestAxionXrayWindow` components. You could perfectly define a grid with vacuum in between strips, of course we do not want that, but in principle one could. You need to add an Al foil so that it is not vacuum.

Then, the idea is that inside the process one will select a combination of structures and/or foils to create a more complex window.